types/dropRepeatsBy.d.ts

```diff
@@ -0,0 +1,3 @@
+export function dropRepeatsBy<T>(fn: (a: T) => T, list: readonly T[]): T[];
```

Comment:
I'm not sure this type is quite right, `fn` doesn't need to return the same type as `list` does it? Maybe something like `dropRepeatsBy<T, U>(fn: (a: T) => U, list: readonly T[]): T[];`

Reply:
Oh yeah, you're totally right. Your suggestion seems like the way to go!

Reply:
Thanks, looks good, I'll leave this up for another day or so and then merge if no one has any further comments.

Reply:
Great, sounds good!